File: 003_msg_publisher_subscriber/scripts/subscriber.py
#!/usr/bin/env python
import rospy
from std_msgs.msg import Float64

def callback(data):
    rospy.loginfo("the number is: %f", data.data)

def main():
    rospy.init_node("subscriber_py",anonymous=True)
    rospy.Subscriber("float_number",Float64,callback)
    rospy.spin()

if __name__ == "__main__":
    try:
        main()
    except rospy.ROSInterruptException:
        pass

File: 003_msg_publisher_subscriber/src/subscriber.cpp
#include <ros/ros.h>
#include <std_msgs/Float64.h>

void callback(const std_msgs::Float64::ConstPtr &msg)
{
    ROS_INFO("Subscribed number is : %f",msg->data);
}

int main(int argc,char **argv)
{
    ros::init(argc,argv,"subscriber_cpp");
    ros::NodeHandle nh;

    ros::Subscriber sub;

    sub = nh.subscribe<std_msgs::Float64>("float_number",100,callback);

    ros::spin();

    return 0;
}